### ISSUE TITLE
Alias subexpressions

### ISSUE THREAD

**Issue by giggio** [nushell/nushell#9732]
### Related problem

I'd like to be able to alias a dynamic value using subexpressions.

### Describe the solution you'd like

For example:

```nu
alias nn = ($env.HOME | path join bin n)
```

### Describe alternatives you've considered

Hard code the alias, but that would not allow my script to be used in other computers where my home differs, or by other people.

**Comment by hardfau18** [comment]
This feature is recently removed ( i don't know exact reason), now alias can't have pipes in it.

 alternative would be to define a function like
```
def nn [] {
$env.HOME | path join bin n
}
```

**Comment by mksinicus** [comment]
`do $closure` will do

```nushell
alias nn = do {$env.HOME | path join bin n}
```

**Comment by giggio** [comment]
@mksinicus 

> `do $closure` will do
> 
> ```nushell
> alias nn = do {$env.HOME | path join bin n}
> ```

I just tried that, but when why run `nn --help` it is different from when I run `$HOME/bin/n --help`, it shows me help from `do`.
And when I run `nn` directly it just prints the path (expanded `/home/<user>/bin/n`).

**Comment by mksinicus** [comment]
@giggio 
> I just tried that, but when why run `nn --help` it is different from when I run `$HOME/bin/n --help`, it shows me help from `do`. And when I run `nn` directly it just prints the path (expanded `/home/<user>/bin/n`).

I see! So you wanted an "[escape to the system](https://www.nushell.sh/book/escaping.html)", not aliasable subexpressions.

TL;DR: You can use

```nushell
alias nn = ^($env.HOME | path join bin n)
```

That caret (^) interprets the string that follows it as a given path to an external call. Similarly `^$env.VISUAL` starts your default visual editor in the current environment. (Unfortunately the usage of caret seems poorly documented in the link above)

**Comment by giggio** [comment]
Yes, that works. I don't get what the difference is, though.
For example, I aliased `terraform` to `tf` and it works just fine:

```nu
alias tf = terraform
```

I also noticed I can alias to the full path:

```nu
alias tf2 = /path/to/terraform
```

Then why do I need the `^` in the subexpression?

Also, I believe my original question is resolved, but I also think this information should be in the docs, as this is a common requirement, to alias to a dynamic path, like a subdir in the home directory.

**Comment by giggio** [comment]
@mksinicus I changed the title and description to make this an issue about docs.

**Comment by amtoine** [comment]
@giggio 
when you say "*an issue about docs*", is that about the help pages of commands like `alias` or the book? :yum:

**Comment by amtoine** [comment]
maybe this issue should be moved to https://github.com/nushell/nushell.github.io?

**Comment by giggio** [comment]
> maybe this issue should be moved to [nushell/nushell.github.io](https://github.com/nushell/nushell.github.io)?

@amtoine yes, I agree. Is there a way to move automatically or we close it here and reopen it there?
